posts/cellular-neighborhoods/cellular-neighborhoods.qmd

```diff
@@ -0,0 +1,270 @@
+
+# the output is a sparse matrix of cells * cells:
+str(neighbors.nearest50)
+
+# compare the number of neighbors found by each approach:
+summary(Matrix::rowSums(neighbors.nearest50))
+summary(Matrix::rowSums(neighbors.radiusbased))
+```
+
+### Subsampling neighbors to minimize spatial auto-correlation
+
+An occasionally important detail: 
+
+In "Mitigating autocorrelation during spatially resolved transcriptomics data analysis", @Maher2023 describe an inconvenient tendency of spatial context matrices: because neighboring cells have largely the same neighbors, their entries in the spatial context matrix are correlated. 
+This correlation between neighbors proves a substantial barrier to distance-based analyses like UMAP or Leiden clustering, producing UMAPs where all points fall in a highly-connected blob and generally poor Leiden performance. 
+(However, for most analyses, correlation between neighboring cells' spatial context vectors doesn't seem to have much impact.)
+They propose that by defining each cell's neighborhood as a random subset of its nearest neighbors, they can
+largely break this correlation between neighbors. They released a python toolkit for this, [SPIN](https://github.com/wanglab-broad/spin){target="_blank"}.
+
+For R coders, here's how you would get a neighborhood matrix with random subsetting:
+
+```{r}
+subsetted_neighbors <- subsampleNeighborsByRow(neighbors = neighbors.nearest50, p = 0.5)
+summary(Matrix::rowSums(subsetted_neighbors))
+```
+
+
+## Summarizing a cell's neighbors to define its spatial context
+
+Usually, you'll employ one of two approaches:
+
+1. Report the average expression of neighborhood cells
+2. Report the cell type abundances within the neighborhood cells
+
+But more bespoke options are possible. For example, you could:
+
+- Only record expression of known ligands, under the theory that they're mainly responsible for cell-cell communication. 
+- Only record genes from a pathway of interest
+- Record QC metrics like the rate of flagged cells, or total counts per cell, or total negprobe counts per cell. 
+- Create a hybrid matrix including both cell type abundances and expression of selected genes. 
+- Instead of computing means, look at SD or covariance of gene expression within a neighborhood. 
+
+The main takeaway here is that once you've defined cellular neighborhoods, it's incredibly
+simply to extract all manner of variables from them, giving you great flexibility in how 
+you pose biological questions. 
+
+You can implement the basic formats of spatial context matrices as follows:
+
+```{r}
+# mean neighborhood expression:
+spatialcontext_expression <- get_neighborhood_expression(counts, neighbors.nearest50)
+# mean cell type abundances:
+spatialcontext_celltypes <- neighbor_tabulate(annot$celltype, neighbors.nearest50)
+
+# spatial context matrices are dense:
+str(spatialcontext_expression)
+str(spatialcontext_celltypes)
+```
+
+
+# Analyzing the spatial context matrix
```

Comment:
For this section, it would be nice to show some visual examples with example code. Not mission critical; nice to have.

Reply:
good idea